ion/src/main/java/com/fasterxml/jackson/dataformat/ion/ionvalue/IonValueDeserializer.java

```diff
@@ -53,4 +53,21 @@ public IonValue deserialize(JsonParser jp, DeserializationContext ctxt) throws I
         throw JsonMappingException.from(jp, "Cannot deserialize embedded object type "
                 + embeddedObject.getClass().getCanonicalName() + " into IonValue");
     }
+
+    @Override
+    public IonValue getNullValue(DeserializationContext ctxt) throws JsonMappingException {
+        try {
+            Object embeddedObj = ctxt.getParser().getEmbeddedObject();
```

Comment:
My spider sense is tingling... am not confident this will reliably work. But maybe it does; `instanceof` acts as `null` check.

But one thing that will definitely be needed is override of:

```
    @Override
    public AccessPattern getNullAccessPattern() {
       return AccessPattern.DYNAMIC;
    }
```

as otherwise null value may be cached (default is `AccessPattern.STATIC`) and so all further deserializations might use the cached value.

Reply:
Thanks @cowtowncoder. I updated the branch with this change.